### ISSUE TITLE
No audit for housekeeping

### ISSUE THREAD

**Issue by escoand** [alerta/alerta#1449]
**Issue Summary**
We have some events that reoccur on every check and some (log entries) have a unique timestamp within the event name to not override them. Now I've the problem that all our reoccurred events are gone and I'm not able to reconstruct how this could happen. I suppose the housekeeping deleted the alerts but I'm not able to verify this.

The configuration looks like this:
```python
# Audit Log
AUDIT_TRAIL = [
    "admin",
    "auth",
    "write",
]  # possible categories are "admin", "write", and "auth"
AUDIT_LOG = True  # set to True to log to application logger
AUDIT_LOG_JSON = True  # log alert data as JSON object
# Housekeeping settings
DEFAULT_EXPIRED_DELETE_HRS = 7 * 24  # hours (0 hours = do not delete)
DEFAULT_INFO_DELETE_HRS = 7 * 24  # hours (0 hours = do not delete)
```

Here is the log of the router and audit trail which doesn't show what happened exactly:
```
Jan 18, 2021 02:02:09.675  APP  [2021-01-18 02:02:09,674] INFO in audit: {"id": "9ee784ae-2491-420e-bed3-1172ce959a1b", "@timestamp": "2021-01-18T01:02:09.674Z", "event": "alert-received", "category": "write", "message": "", "user": {"id": "alerta", "customers": [], "scopes": ["admin", "write", "read"]}, "resource": {"id": "99b82b0a-ac4e-4509-8435-937f62eff8e0", "type": "alert"}, "request": {"id": "d9149c03-2fe9-4d7e-aa45-f98b34f414c2", "endpoint": "api.receive", "method": "POST", "url": "https://.../api/alert", "args": {}, "data": {"customer": "...", "environment": "...", "service": ["..."], "resource": "...", "origin": "...", "event": "...", "value": "...", "rawData": "...", "severity": "major", "timeout": 86400, "attributes": {}}, "ipAddress": "192.168.153.236"}, "extra": {}}
Jan 18, 2021 03:00:13.466  RTR  200 GET /api/alert/99b82b0a-ac4e-4509-8435-937f62eff8e0/notes (45.807 ms)
Jan 18, 2021 03:00:13.468  RTR  200 GET /api/alert/99b82b0a-ac4e-4509-8435-937f62eff8e0 (54.882 ms)
Jan 18, 2021 03:04:08.933  APP  [2021-01-18 03:04:08,932] INFO in audit: {"id": "9ef4cbbf-421d-413c-8d31-16b47268d54f", "@timestamp": "2021-01-18T02:04:08.931Z", "event": "alert-note-added", "category": "write", "message": "", "user": {"id": "...", "customers": [], "scopes": ["admin", "read", "write"]}, "resource": {"id": "a115c91f-980b-47c9-b2cd-14c186c9999a", "type": "note"}, "request": {"id": "168e127f-b134-4b49-985f-219c674027f7", "endpoint": "api.add_note", "method": "PUT", "url": "https://.../api/alert/99b82b0a-ac4e-4509-8435-937f62eff8e0/note", "args": {}, "data": {"text": "..."}, "ipAddress": "192.168.153.236"}, "extra": {}}
Jan 18, 2021 03:04:08.937  RTR  201 PUT /api/alert/99b82b0a-ac4e-4509-8435-937f62eff8e0/note (41.432 ms)
Jan 18, 2021 03:04:09.378  RTR  200 GET /api/alert/99b82b0a-ac4e-4509-8435-937f62eff8e0/notes (39.188 ms)
Jan 18, 2021 03:04:11.068  APP  [2021-01-18 03:04:11,067] INFO in audit: {"id": "6555e8d7-c7e3-415e-8cba-296260543a36", "@timestamp": "2021-01-18T02:04:11.066Z", "event": "alert-actioned", "category": "write", "message": "close operator action", "user": {"id": "...", "customers": [], "scopes": ["admin", "read", "write"]}, "resource": {"id": "99b82b0a-ac4e-4509-8435-937f62eff8e0", "type": "alert"}, "request": {"id": "ab843f10-a5d0-4427-9ba5-2170a1352732", "endpoint": "api.action_alert", "method": "PUT", "url": "https://.../api/alert/99b82b0a-ac4e-4509-8435-937f62eff8e0/action", "args": {}, "data": {"action": "close"}, "ipAddress": "192.168.153.236"}, "extra": {}}
Jan 18, 2021 03:04:11.072  RTR  200 PUT /api/alert/99b82b0a-ac4e-4509-8435-937f62eff8e0/action (49.509 ms)
Jan 18, 2021 03:04:11.157  RTR  200 GET /api/alert/99b82b0a-ac4e-4509-8435-937f62eff8e0 (36.479 ms)
Jan 20, 2021 07:21:26.772  RTR  200 GET /alert/99b82b0a-ac4e-4509-8435-937f62eff8e0 (1.829 ms)
Jan 20, 2021 07:21:27.435  RTR  404 GET /api/alert/99b82b0a-ac4e-4509-8435-937f62eff8e0 (56.688 ms)
Jan 20, 2021 07:21:27.446  RTR  404 GET /api/alert/99b82b0a-ac4e-4509-8435-937f62eff8e0/notes (41.809 ms)
```

**Environment**
- OS: Linux
- API version: 8.3.2
- Deployment: Docker
- Database: MongoDB
- Server config:
  Auth enabled? Yes
  Auth provider? OpenID
  Customer views? Yes
- web UI version: 8.4.0

**Expected behavior**
The audit trail should cover the whole life cycle of the alert and maybe also the note.

**Comment by escoand** [comment]
Hmmm, my closed alerts are getting removed with the next housekeeping. Is there some kind of debug logging?

**Comment by satterly** [comment]
Set `LOG_LEVEL='DEBUG'`. See https://docs.alerta.io/en/latest/configuration.html#logging-settings

**Comment by satterly** [comment]
Audit logging of housekeeping was added in #1240 and included in version 8.0.0 released in June 2020.
